### DISCUSSION TITLE
Directory.glob: make pattern optional

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12143] [Ideas]
`Directory.glob()` should be callable without arguments. The intuitive default for `pattern` is `**`, which will match everything.

**Comment by jedevc** [comment]
Making it optional means that the standard way of calling it would now require an opts struct - given that's probably the general case, I think it makes more sense to optimize that case for simplicity.

(note, this would also be a breaking change)

**Comment by shykes** [comment]
> Making it optional means that the standard way of calling it would now require an opts struct

Only in Go :)

> (note, this would also be a breaking change)

Only in Go :)
